raiden/tests/api/test_api.py

```diff
@@ -80,6 +105,71 @@ def test_channel_to_api_dict(raiden_network, token_addresses, settle_timeout):
     assert result == expected_result
 
 
+def test_url_with_invalid_address(rest_api_port_number, api_backend):
+    """ Addresses required the leading 0x in the urls. """
+
+    url_without_prefix = (
+        'http://localhost:{port}/api/1/'
+        'channels/ea674fdde714fd979de3edf0f56aa9716b898ec8'
+    ).format(port=rest_api_port_number)
+
+    request = grequests.patch(
+        url_without_prefix,
+        json={'state': CHANNEL_STATE_SETTLED}
+    )
+    response = request.send().response
+
+    assert response.status_code == httplib.NOT_FOUND
```

Comment:
URL is universal resource locator, a bad URL means the resource does not exist, also this is controlled by flask.